### PR TITLE
Azure scenario-runner cleanup

### PR DESCRIPTION
**Pull request by znewton**
## Description

Follow-up on #15057 addressing feedback from @ssimic2.

## Breaking Changes

Scenario-runner is experimental now, after #15057, so this should be OK.

1. All connection and service auth information comes from Env now, instead of each individual scenario config.
2. All *RunnerClient files have been consolidated into `scenarioRunnerClient.ts`, so running via client file would now be `./dist/scenarioRunnerClient.js <ScenarioNameRunner> [...args]`

## Reviewer Guidance

My goal here is to reduce the amount of boilerplate required to create more runners for various scenarios. Please let me know if this is a good start or if I went overboard on deduping common configs.

Also, AzureClientRunner is kinda weird and doesn't fit the pattern of the other Runners. It is likely **always** going to be the first stage in any scenario config. Can we just make it a function that happens in TestOrchestrator?

### CODE REVIEW

File: azure/packages/test/scenario-runner/src/ScenarioRunner.ts [hunk 1]
```diff
@@ -0,0 +1,166 @@
...
+	IRunner,
+	IRunnerEvents,
+	IRunnerStatus,
+	IScenarioConfig,
+	IScenarioRunConfig,
+	RunnerStatus,
+} from "./interface";
+import { convertConfigToScriptParams, createChildProcess } from "./utils";
+
+export abstract class ScenarioRunner<
```
Comment (ssimic2) [inline azure/packages/test/scenario-runner/src/ScenarioRunner.ts:18]: Love it! Simplifies life of other runners. They focus on core scenarios they are describing and not mechanics of running tests.

File: azure/packages/test/scenario-runner/src/DocLoaderRunner.ts [hunk 2]
```diff
@@ -231,35 +106,46 @@ export class DocLoaderRunner extends TypedEventEmitter<IRunnerEvents> implements
...
+		const scenarioRunConfig: DocLoaderRunConfig = {
+			...runConfig,
+			childId: options.childId,
+			docId: options.docId,
+			schema: this.scenarioConfig.schema,
+			client: this.scenarioConfig.client,
+		};
+		if (!options.isSync) {
+			delete scenarioRunConfig.logger;
+			delete scenarioRunConfig.client;
```
Comment (ssimic2) [inline azure/packages/test/scenario-runner/src/DocLoaderRunner.ts:143]: I get why you are doing it, just curious if there is maybe a cleaner way. I have not looked in-depth the full path, but in async scenarios we maybe need to make sure that client/logger are never passed down. This is minor anyways.

**Comment by ssimic2** [comment]
And I agree that AzureClientRunner is now somewhat weird. Maybe questions to help reason if to keep it: obvious one - are we still using it? If not, do we anticipate scenarios where it would be valuable to move it across stages (create/loads across same azure client instance)? Is it an antipatern in the way user reasons about the tests (say they want to add their own runner stage)?

In uncertain, either way, we will find some answers as we explore concrete usage.

**Review by ssimic2 — APPROVED** [review]

**Comment by ssimic2** [comment]
And loved the changes you made here!

**Comment by znewton** [comment]
Going to do a follow-up PR for these comments:

1. AzureClientRunner
> And I agree that AzureClientRunner is now somewhat weird. Maybe questions to help reason if to keep it: obvious one - are we still using it? If not, do we anticipate scenarios where it would be valuable to move it across stages (create/loads across same azure client instance)? Is it an antipatern in the way user reasons about the tests (say they want to add their own runner stage)?
> 
> In uncertain, either way, we will find some answers as we explore concrete usage.

2. buildScenarioRunConfig()
> I get why you are doing it, just curious if there is maybe a cleaner way. I have not looked in-depth the full path, but in async scenarios we maybe need to make sure that client/logger are never passed down. This is minor anyways.

**Pull request merged**